packages/react-toast/src/hooks.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from 'react';
+import {Toast} from '@shopify/polaris';
+
+export const ToastContext = React.createContext<ToastActions>({
+  show: noop,
+  hide: noop,
+});
+
+export interface ToastDescription {
+  content: string;
+  error?: boolean;
+  dismissible?: boolean;
+  duration?: number;
+  onDismiss?(): void;
+}
+
+export interface ToastActions {
+  show(options: ToastDescription): void;
+  hide(): void;
+}
+
+export interface Props {
+  children: any;
+}
+
+export function ToastProvider({children}: Props) {
+  const [toasts, setToasts] = React.useState<ToastDescription[]>([]);
+
+  const show = (toast: ToastDescription) => {
```

Comment:
We had talked about supporting the declarative API as well, should the context maybe be like 

```
{
  show(toast: ToastDescription): void;
  hide(): void;
  Toast: ReactComponent<ToastDescription, never>;
}
```

so you can do

```
const {Toast} = useToast()

return (
  /* rest of render*/
 {error && <Toast /*some props*/ /> 
);

```